python/ray/air/dataset_iterator.py

```diff
@@ -0,0 +1,199 @@
+import abc
```

Comment:
@bveeramani I decided to leave this in AIR for now since it has ML-specific iterators.

Reply:
Okay, sounds good